src/main/java/app/domain/lotto/Lotto.java

```diff
@@ -0,0 +1,89 @@
+package app.domain.lotto;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class Lotto {
+    private static final int LOTTO_SIZE = 6;
+
+    private List<LottoNum> numbers;
+
+    private Lotto(List<LottoNum> numbers) {
+        List<LottoNum> inputList = new ArrayList<>(numbers);
+        validation(inputList);
+        this.numbers = inputList;
+    }
+
+    public static Lotto from(List<Integer> list) {
+        return new Lotto(list.stream()
+                .map(LottoNum::of)
+                .collect(Collectors.toList()));
+    }
+
+    public static Lotto from(String list) {
+        return new Lotto(Arrays.stream(list.split(","))
+                .map(str -> LottoNum.of(str.trim()))
+                .collect(Collectors.toList()));
+    }
+
+    private void validation(List<LottoNum> list) {
+        if (list.size() != LOTTO_SIZE) {
+            throw new IllegalArgumentException(String.format("로또 길이는 %d개 이어야 합니다.", LOTTO_SIZE));
+        }
+        if (isDuplicate(list)) {
+            throw new IllegalArgumentException("숫자가 중복 되어있습니다.");
+        }
+    }
+
+    private boolean isDuplicate(List<LottoNum> list) {
+        Set<LottoNum> set = new HashSet<>(list);
+        return set.size() != LOTTO_SIZE;
+    }
+
+    public List<LottoNum> getNumbers() {
+        arrangeNumbers();
+        return new ArrayList<LottoNum>(this.numbers);
+    }
+
+    private void arrangeNumbers() {
+        this.numbers = numbers.stream().sorted(LottoNum::compareTo).collect(Collectors.toList());
+    }
+
+    public int matchLotto(List<LottoNum> winnerNumberList) {
```

Comment:
```suggestion
    public int matchLotto(List<LottoNum> winnerNumbers) {
```

List와 같은 자료형을 변수명으로 사용하면 어떠한 단점이 있을지 고민해보아요